internal/api/v1/application/update.go

```diff
@@ -288,6 +289,12 @@ func updateRoutes(
 
 	routes := []string{}
 	for _, d := range updateRoutes {
+		// Strip scheme prefixes, if present
+		routeURL, err := url.Parse(d)
+		if err == nil && routeURL.Scheme != "" {
+			d = strings.TrimPrefix(d, routeURL.Scheme+"://")
+		}
```

Comment:
Same, I would add the err check

```suggestion
		routeURL, err := url.Parse(d)
		if err != nil {
			return apierror.NewBadRequestError(err.Error()).WithDetails("failed to parse route")
		}
		if routeURL.Scheme != "" {
			d = strings.TrimPrefix(d, routeURL.Scheme+"://")
		}
```

Reply:
ok, commit e5c46e789fa2ccdeb3fd9554f74658ca5bce21e4